src/recordlinker/routes/person_router.py

```diff
@@ -170,3 +170,39 @@ def merge_person_clusters(
         service.delete_persons(session, persons, commit=False)
 
     return schemas.PersonRef(person_reference_id=person.reference_id)
+
+
+@router.delete(
+    "/{person_reference_id}",
+    summary="Delete an empty Person",
+    status_code=fastapi.status.HTTP_204_NO_CONTENT,
+)
+def delete_empty_person(
+    person_reference_id: uuid.UUID,
+    session: orm.Session = fastapi.Depends(get_session),
+):
+    """
+    Delete an empty Person from the MPI database.
+    """
+    # Check that person_reference_id is valid
+    person = service.get_person_by_reference_id(session, person_reference_id)
+
+    if person is None:
+        raise fastapi.HTTPException(status_code=fastapi.status.HTTP_404_NOT_FOUND)
+
+    # Check if the person has associated patients
+    if person.patients:
```

Comment:
I'm 99% sure this is fine and is preferable because it reads well, however as is this retrieves all the linked patients out of the database.  We really only need to run
```sql
SELECT 1 
FROM mpi_patient 
WHERE person_id = X 
LIMIT 1;
```

Can you think of any situations where the person cluster would be so large, multiple thousands, where it be advantageous to just to a quick existence check rather than retrieve all?

Reply:
That's a good point! From our experience in LAC, most person clusters were n~13 so definitely not getting up there. I modified the query to match your suggestion.